addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -69,18 +83,19 @@ def _work_time_per_day(self):
                     if date_to > utc.localize(leave.date_from) and date_from < utc.localize(leave.date_to):
                         tmp_start = max(date_from, utc.localize(leave.date_from))
                         tmp_end = min(date_to, utc.localize(leave.date_to))
-                        results[leave.id][tmp_start.date()] += (tmp_end - tmp_start).total_seconds() / 3600
-                results[leave.id] = sorted(results[leave.id].items())
+                        results[calendar_id][leave.id][tmp_start.date()] += (tmp_end - tmp_start).total_seconds() / 3600
+                results[calendar_id][leave.id] = sorted(results[calendar_id][leave.id].items())
         return results
 
     def _timesheet_create_lines(self):
         """ Create timesheet leaves for each employee using the same calendar containing in self.calendar_id
 
             If the employee has already a time off in the same day then no timesheet should be created.
         """
-        work_hours_data = self._work_time_per_day()
+        resource_calendars = self._get_resource_calendars()
```

Comment:
Not needed anymore :)

Reply:
Not done as well, I guess it is because the `resource_calendars` is also used in the read_group. 🙂 🤷🏼‍♂️